dbms/src/Functions/FunctionsConversion.h

```diff
@@ -1656,6 +1657,14 @@ class FunctionCast final : public IFunctionBase
                 ConvertImpl<LeftDataType, RightDataType, NameCast>::execute(block, arguments, result, input_rows_count, scale);
                 return true;
             });
+
+            /// Additionally check if callOnIndexAndDataType wasn't called at all.
+            if (!res)
```

Comment:
There’s no way to get into this if statement now. Upper check for supported type forces the function return true. It’s better to throw logical error here.

Reply:
Yes. But let's still check it just in case if somebody add new type or something.